File: Data Structures & Algorithms/longest-common-prefix/submission-0.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        s = strs[0]
        for i in range(1,len(strs)):
            j = 0
            while j<len(s) and j<len(strs[i]) and s[j] == strs[i][j]:
                j += 1
            s = s[:j]
            if not s:
                return ""
        return s
        

File: Data Structures & Algorithms/string-encode-and-decode/submission-0.java
class Solution {
    //LeetCode 271
    public String encode(List<String> strs) {
        StringBuilder s = new StringBuilder();
        for(String st:strs){
            s.append(st.length()).append('#').append(st);
        }
        return s.toString();
    }

    public List<String> decode(String str) {
        int n = str.length();
        int i = 0;
        List<String> arr = new ArrayList<>();
        while(i<n){
            int j = i;
            while(str.charAt(j) != '#') j++;
            int length = Integer.parseInt(str.substring(i,j));
            j++;
            arr.add(str.substring(j,j+length));
            i = j+length;
        }
        return arr;
    }
}
